components/bitbucket-api-client/src/client.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::Read;
+
+use hab_http::ApiClient;
+use oauth_common::{OAuthError, OAuthResult};
+use oauth_common::types::*;
+use hyper::header::{Authorization, Basic, Bearer, ContentType};
+use serde_json;
+
+use config::BitbucketCfg;
+use types::*;
+
+#[derive(Clone)]
+pub struct BitbucketClient {
+    pub config: BitbucketCfg,
+}
+
+impl BitbucketClient {
+    pub fn new(config: BitbucketCfg) -> Self {
+        BitbucketClient { config: config }
+    }
+}
+
+impl OAuthClient for BitbucketClient {
+    // This function takes the code received from the Oauth dance and exchanges
+    // it for an access token
+    fn authenticate(&self, code: &str) -> OAuthResult<String> {
+        // TODO JB: make the version here dynamic
+        let client = ApiClient::new(&self.config.web_url, "habitat", "0.54.0", None)
```

Comment:
would be awesome if possible to not put in an actual version until we have dynamic versions working as it will be quickly out of date/inaccurate

Reply:
It's not optional, but I'm open to suggestions.  I can also try to get the dynamic versions working.  I'm honestly not sure how big of a headache that will be.